Adding notes about providing code changes and pull requests. (#95)
StoglRobotics/ros_team_workspace: docs/guidelines/development.rst

**NOTE**: All the proposal here are the results of authors' personal experiences. Saying that, if you have any idea to make them better you are very wellcome to create a PR.

Overview
=========

1. `Making changes to the code-base by submitting a Pull Request <#>`_

2. `Using *pre-commit* <#use-pre-commit-for-formatting-and-linting>`_

Code Changes and Pull Request Submissions
==========================================

> To make a change of the existing code base Pull Request (GitHub) or Merge Request (GitLab) are used.

Reviewer: To make a change of the existing code base, Pull Request (GitHub) or Merge Request (GitLab) are used.

ADDING COMMA WILL BE MORE CLEAR

> To make a change of the existing code base Pull Request (GitHub) or Merge Request (GitLab) are used.

Reviewer: To make a change of the existing code base, Pull Request (GitHub) or Merge Request (GitLab) are used.

Author: ```suggestion
To make a change of the existing code base, Pull Request (GitHub) or Merge Request (GitLab) are used.
```

> This section describe in short process with valuable tips to make your and reviewers' life easier.

Reviewer: This section describes in

Author: ```suggestion
This section describes in short process with valuable tips to make your and reviewers' life easier.
```

      This simplify collaboration inside the team - there is no need for individual access grants when using forks under your user.
      **Always** check if there is already a fork in *StoglRobotics-forks* organization and if not create it.

#. Start development always from the up-to date state of the repositories default branch (usually called *master* or *main* - for simplicity we call it *master* here).
   Take into account that master branch of your fork is usually not up to date with the upstream repository.
   Therefore be careful about that and use the opportunity to sync *master* branch of the fork to the state of the *master* branch of the upstream repository.

#. **Always** create a new branch for each feature or bug fix. Don't submit PRs from *master* branch.

#. **Before** starting development check how the branch will be merged, using *merge commit* or *squash* method.
   If *merge commit* is used make sure that each of your commits is clean and named properly since they will become part of the repository's history.

   .. note::

>       In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.

Reviewer: `In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.`

>       In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.

Reviewer: `In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.`

>       In *Stogl Robotics* we are always *squashing* commits, i.e., one future or one bug fix is one commit in the default branch. The commit message is edited before merging.

Reviewer: `In *Stogl Robotics* we are always *squashing* commits, i.e., one feature or one bug fix is one commit in the default branch. The commit message is edited before merging.`

> #. Explain in the PR/MR description what you code is doing and why.

Reviewer: Explain in the PR/MR description what your

Author: ```suggestion
#. EExplain in the PR/MR description what your code is doing and why.
```

#. When you are finished with development and want to submit code for the review - consider the following tips:

   - **Always** run *pre-commit* formaters;
   - Review your code **first by yourself** before asking someone else;

>    - Make sure there **are not** commended code blocks or if they have to be there add explanation why;

Reviewer: - Make sure there **are not** commented code
